testing/jormungandr-testing-utils/src/testing/node/logger.rs

```diff
@@ -257,35 +305,42 @@ impl JormungandrLogger {
         if let Ok(result) = legacy_entry {
             return Ok(result.into());
         }
-        // parse and aggregate spans fields
-        let mut jsonize_entry: serde_json::Value = serde_json::from_str(&line)?;
-        let mut aggreagated: HashMap<String, serde_json::Value> = HashMap::new();
-        if let Some(fields) = jsonize_entry.get("fields") {
-            // main span "fields" is ensured be an object, it is safe to unwrap here
-            for (k, v) in fields.as_object().unwrap().iter() {
-                aggreagated.insert(k.clone(), v.clone());
-            }
+
+        // Fields could be of various types, since we do not need to modify or
+        // interact with them, it is easier to just map them to strings
+        macro_rules! stringify_map {
+            ($container:expr, $field:expr) => {
+                let mapped = $container
+                    .index($field)
+                    .as_object()
+                    .expect("not an object")
+                    .into_iter()
+                    .map(|(k, v)| {
+                        (
+                            k,
+                            if v.is_string() {
+                                v.to_owned()
+                            } else {
+                                serde_json::Value::String(v.to_string())
+                            },
+                        )
+                    })
+                    .collect();
+                *$container.index_mut($field) = mapped;
+            };
```

Comment:
Could this be a function instead? I don't see the advantage of this being a macro. 
What am I missing? 🤔